core/src/main/scala/tech/beshu/ror/utils/MatcherWithWildcards.java

```diff
@@ -14,141 +14,84 @@
+        this.matchers = Sets.newHashSet(patterns);
+        for (String p : patterns) {
+            if (p == null) {
+                continue;
+            }
+            patternsList.add(p.split("\\*+", -1 /* want empty trailing token if any */));
+        }
     }
 
-    int idx = pattern[0].length();
-    for (int i = 1; i < pattern.length - 1; ++i) {
-      String patternTok = pattern[i];
-      int nextIdx = line.indexOf(patternTok, idx);
-      if (nextIdx < 0) {
-        return false;
-      }
-      else {
-        idx = nextIdx + patternTok.length();
-      }
+    public Set<String> getMatchers() {
+        return matchers;
     }
 
-    return line.endsWith(pattern[pattern.length - 1]);
-  }
-
-  public Set<String> getMatchers() {
-    return matchers;
-  }
-
-  public boolean match(String haystack) {
-    if (haystack == null) {
-      return false;
-    }
-    for (String[] p : patternsList) {
-      if (miniglob(p, haystack)) {
-        return true;
-      }
+    public boolean match(T haystack) {
+        if (haystack == null) {
+            return false;
+        }
+        for (String[] p : patternsList) {
+            if (miniglob(caseMappingEquality, p, haystack)) {
+                return true;
+            }
+        }
+        return false;
     }
-    return false;
-  }
 
-  public boolean match(boolean remoteClusterAware, String haystack) {
-    return remoteClusterAware ? matchRemoteClusterAware(haystack) : match(haystack);
-  }
-
-  public Set<String> filter(boolean remoteClusterAware, Set<String> haystack) {
-    return remoteClusterAware ? filterRemoteClusterAware(haystack) : filter(haystack);
-  }
-
-  public boolean matchRemoteClusterAware(String haystack) {
-    if (haystack == null) {
-      return false;
-    }
+    private static <T> boolean miniglob(CaseMappingEqualityJava<T> caseMappingEquality, String[] pattern, T line) {
+        String showedLine = caseMappingEquality.mapCases(caseMappingEquality.show(line));
+        if (pattern.length == 0) {
+            return Strings.isNullOrEmpty(showedLine);
+        } else if (pattern.length == 1) {
+            return showedLine.equals(caseMappingEquality.mapCases(pattern[0]));
+        }
+        if (!showedLine.startsWith(caseMappingEquality.mapCases(pattern[0]))) {
+            return false;
+        }
 
-    boolean remoteClusterRequested = haystack.contains(":");
+        int idx = pattern[0].length();
+        for (int i = 1; i < pattern.length - 1; ++i) {
+            String patternTok = caseMappingEquality.mapCases(pattern[i]);
+            int nextIdx = showedLine.indexOf(patternTok, idx);
+            if (nextIdx < 0) {
+                return false;
+            } else {
+                idx = nextIdx + patternTok.length();
+            }
+        }
 
-    for (String[] p : patternsList) {
-      // Ignore remote cluster related permissions if request didn't mean it
-      if (!remoteClusterRequested && Arrays.asList(p).contains(":")) {
```

Comment:
if haystack hasn't colon,and pattern has colon, miniglob won't match pattern, so `if(){continue}` expression doesn't change behaviour